assembl/static2/js/app/components/common/treeItem.jsx

```diff
@@ -0,0 +1,267 @@
+        // if listRef.Grid is null, it means it has been unmounted, so we are now on a new List
+        if (listRef.Grid) {
+          // In Firefox (tested on version 59), recomputing row heights can jump back the page scroll
+          // to the same post (The scrollTop from the Grid component is fine.),
+          // potentially a post with a youtube video, but may be a coincidence.
+          // Saving pageYOffset and restoring it after recomputeRowHeights fixes the issue.
+          const pageYOffset = window.pageYOffset;
+          listRef.recomputeRowHeights(minRowIndex);
+          window.scrollTo({ top: pageYOffset, left: 0 });
+          nuggetsManager.update();
+          // recompute height only for rows (top post) starting at rowIndex
+        }
+        minRowIndex = null;
+        timeoutId = null;
+      }, delay);
+    }
+  };
+
+  expandCollapse = (event: SyntheticEvent<HTMLDivElement>) => {
+    event.stopPropagation();
+    this.setState(
+      state => ({ expanded: !state.expanded }),
+      () => {
+        this.resizeTreeHeight(0);
+      }
+    );
+  };
+
+  renderToggleLink = (expanded: boolean, indented: boolean) => (
+    <div
+      ref={this.scrollAnchor}
+      onClick={(event) => {
+        if (expanded) {
+          this.scrollToElement();
+        }
+        this.expandCollapse(event);
+      }}
+      className={indented ? 'expand-indented' : 'expand'}
+    >
+      {expanded ? <span className="assembl-icon-minus-circled" /> : <span className="assembl-icon-plus-circled" />}
+    </div>
+  );
+
+  scrollToElement = () => {
+    scrollToPost(this.scrollAnchor);
+  };
+
+  render() {
+    const {
+      id,
+      children,
+      identifier,
+      phaseId,
+      lang,
+      originalLocale,
+      fullLevel,
+      level,
+      hidden,
+      rowIndex, // the index of the row (i.e. level 0 item) in the List
+      contentLocaleMapping,
+      InnerComponent,
+      InnerComponentFolded,
+      SeparatorComponent,
+      nuggetsManager,
+      listRef,
+      cache
+    } = this.props;
+    const { expanded, visible } = this.state;
+    const numChildren = children ? children.length : 0;
+    const contentLocale = contentLocaleMapping.getIn([id, 'contentLocale'], originalLocale);
+    const forwardProps = {
+      contentLocale: contentLocale,
+      ...this.props,
+      numChildren: numChildren
+    };
+    delete forwardProps.children;
+    // InnerComponent, the post, is only rendered when the Child appears in the viewport or next page
+    const { hash } = window.location;
+    let isVisible = visible;
+    // load right away the shared post
+    let hashid;
+    if (hash !== '') {
+      hashid = hash.replace('#', '').split('?')[0];
+      isVisible = hashid === id || isVisible;
+    }
+
+    return (
+      <div
+        className={classnames(`level level-${level}`, {
+          'border-left child-level': level > 0,
+          'no-shift': level > 3,
+          'padding-right': level > 4,
```

Comment:
Could it be useful to set those integer as constants with a name explaining their usefullness so we could understand why and customize them if necessary ?